x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1196,6 +1200,77 @@ protected Expression rule(Expression e) {
             return e.foldable() ? Literal.of(e) : e;
         }
     }
+    
+    static class LocalRelationCountFolding extends OptimizerRule<LogicalPlan> {
+
+        @Override
+        protected LogicalPlan rule(LogicalPlan p) {
+            if (Verifier.isLocalRelation(p)) {
+                // collect aggregate aliases, if any
+                List<Alias> aliases = new ArrayList<>();
+                if (p instanceof Filter) {
```

Comment:
I think this check along with the `instanceof Aggregate` should be done in a more generic way, so that they also catch the queries with Order By and/or Limit.